NanoProd/python/DiMuon_cff.py

```diff
@@ -235,25 +238,53 @@ def defineFiltersAndProducers(isRun2):
+  this.dsaPatTable.variables.highPtId = Var("?passed('CutBasedIdGlobalHighPt')?2:passed('CutBasedIdTrkHighPt')","uint8",doc="high-pT cut-based ID (1 = tracker high pT, 2 = global high pT, which includes tracker high pT)")
+  this.dsaPatTable.variables.pfIsoId = Var("passed('PFIsoVeryLoose')+passed('PFIsoLoose')+passed('PFIsoMedium')+passed('PFIsoTight')+passed('PFIsoVeryTight')+passed('PFIsoVeryVeryTight')","uint8",doc="PFIso ID from miniAOD selector (1=PFIsoVeryLoose, 2=PFIsoLoose, 3=PFIsoMedium, 4=PFIsoTight, 5=PFIsoVeryTight, 6=PFIsoVeryVeryTight)")
+  this.dsaPatTable.variables.tkIsoId = Var("?passed('TkIsoTight')?2:passed('TkIsoLoose')","uint8",doc="TkIso ID (1=TkIsoLoose, 2=TkIsoTight)")
+  # this.dsaPatTable.variables.miniIsoId = Var("passed('MiniIsoLoose')+passed('MiniIsoMedium')+passed('MiniIsoTight')+passed('MiniIsoVeryTight')","uint8",doc="MiniIso ID from miniAOD selector (1=MiniIsoLoose, 2=MiniIsoMedium, 3=MiniIsoTight, 4=MiniIsoVeryTight)")
+  # this.dsaPatTable.variables.mvaMuID = Var("mvaIDValue()",float,doc="MVA-based ID score ",precision=6)
+  # this.dsaPatTable.variables.multiIsoId = Var("?passed('MultiIsoMedium')?2:passed('MultiIsoLoose')","uint8",doc="MultiIsoId from miniAOD selector (1=MultiIsoLoose, 2=MultiIsoMedium)")
+  # this.dsaPatTable.variables.puppiIsoId = Var("passed('PuppiIsoLoose')+passed('PuppiIsoMedium')+passed('PuppiIsoTight')", "uint8", doc="PuppiIsoId from miniAOD selector (1=Loose, 2=Medium, 3=Tight)")
+  this.dsaPatTable.variables.triggerIdLoose = Var("passed('TriggerIdLoose')",bool,doc="TriggerIdLoose ID")
+  this.dsaPatTable.variables.inTimeMuon = Var("passed('InTimeMuon')",bool,doc="inTimeMuon ID")
+  # this.dsaPatTable.variables.jetNDauCharged = Var("?userCand('jetForLepJetVar').isNonnull()?userFloat('jetNDauChargedMVASel'):0", "uint8", doc="number of charged daughters of the closest jet")
+  # time variables
   this.dsaPatTable.variables.rpcTimeInOut = Var('rpcTime().timeAtIpInOut', float, doc='RPC time in out')
   this.dsaPatTable.variables.timeInOut = Var('time().timeAtIpInOut', float, doc='time in out')
   this.dsaPatTable.variables.rpcTimeInOutErr = Var('rpcTime().timeAtIpInOutErr', float, doc='RPC time error in out')
   this.dsaPatTable.variables.timeInOutErr = Var('time().timeAtIpInOutErr', float, doc='time error in out')
   this.dsaPatTable.variables.rpcTimeNdof = Var('rpcTime().nDof', float, doc='RPC time ndof')
   this.dsaPatTable.variables.timeNdof = Var('time().nDof', float, doc='time ndof')
 
+  # track kink
+  this.dsaPatTable.variables.trkKink = Var('combinedQuality().trkKink', float, doc='trkKink')
+
+  # For some variables, save also bestTrack information
+  common_vars = ['chi2', 'ndof', 'dxy', 'dxyErr', 'dz', 'dzErr', 'pt_error', 'n_valid_hits', 'n_lost_hits']
+  for var_name in common_vars:
+    orig_var = getattr(this.dsaPatTable.variables, var_name)
+    final_expr = ''.join(['bestTrack.'] + orig_var.expr.value().split('.')[1:])
```

Comment:
Do we want to store some indicator for the type of best track?

Reply:
I already produced the data samples without this info, so I think it's fine for the moment.

Reply:
agreed, to be kept in mind for a possible future update